frontend/pages/inventory.tsx

```diff
@@ -1,9 +1,147 @@
-import React from 'react';
+import { FC, ReactNode, useState, useEffect } from 'react';
+import { API_URL } from '../config';
+import { toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
 
-function Inventory() {
+interface InventoryLayoutProps {
+    children: ReactNode;
+}
+
+const InventoryLayout: FC<InventoryLayoutProps> = ({ children }) => {
     return (
-        <h1>Inventory</h1>
+        <div className="container mx-auto mt-8">
+            <h1 className="text-4xl font-bold mb-4">Inventory</h1>
+            <div className="overflow-x-auto">{children}</div>
+        </div>
     );
+};
+
+interface InventoryItem {
+    id: number;
+    name: string;
+    description: string;
+    quantity: number;
+}
+
+interface InventoryPageProps {
+    inventoryData: InventoryItem[];
 }
 
-export default Inventory;
+const InventoryPage: FC<InventoryPageProps> = ({ inventoryData }) => {
+    const itemsPerPage = 6;
+    const [currentPage, setCurrentPage] = useState(1);
+
+    const indexOfLastItem = currentPage * itemsPerPage;
+    const indexOfFirstItem = indexOfLastItem - itemsPerPage;
+    const currentItems = inventoryData.slice(indexOfFirstItem, indexOfLastItem);
+
+    const totalPages = Math.ceil(inventoryData.length / itemsPerPage);
+
+    useEffect(() => {
+        // Update URL with page parameter
+        window.history.replaceState({}, '', `/inventory?page=${currentPage}`);
```

Comment:
useHistory()

Reply:
it need a router which we dont have on the other pages either (index nor project)

Reply:
what about something like this ? https://stackoverflow.com/questions/62845014/change-url-without-page-refresh-next-js#62845110